### PR TITLE
F#src 3733 separate teleop product from demo

### PR DESCRIPTION
**Pull request by ToivoS**

### CODE REVIEW

File: bin/run-ansible.sh [hunk 1]
```diff
@@ -156,13 +160,21 @@ if [[ "${playbook}" = "server_and_nuc_deploy" ]]; then
...
+        aurora_inventory="ansible/inventory/teleop/${aurora_inventory}"
+    fi
+
     echo ""
     echo " ---------------------------------------------------"
-    echo " |             SSH and BECOME passwords:           |"
-    echo " | SSH: enter the NUC sudo password                |"
-    echo " | BECOME: just press enter (same as SSH password) |"
+    echo " |                 VAULT password:                 |"
+    echo " | Enter the VAULT password provided by Shadow     |"
```
Comment (biofotis) [inline bin/run-ansible.sh]: so no need to have ssh password anymore? What if I want to use it on my setup here in Bristol?

File: README.md [hunk 2]
```diff
@@ -48,33 +48,48 @@ Molecule user guide is available [here](https://molecule.readthedocs.io/en/stabl
 
 ## teleop_deploy ##
 
-For deploying teleop software on multiple machines (server, control-machine, client, windows-machine)
+For deploying teleop software on multiple machines (server, control-machine, client)
+
+For Hand E teleop software deployments on a laptop (called "server" in this playbook) and a control machine (NUC). If remote_teleop=true, then software will also be deployed on a third computer (called "client").
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply README.md]: Fixed in the last commit

File: README.md [hunk 3]
```diff
@@ -48,33 +48,48 @@ Molecule user guide is available [here](https://molecule.readthedocs.io/en/stabl
...
+
+To begin with, the teleop_deploy playbook checks the installation status of docker. If docker is not installed then a new clean installation is performed. If the required image is private, 
+then a valid Docker Hub account with pull credentials from Shadow Robot's Docker Hub is required. Then the specified docker image is pulled and a docker container is initialized. Finally, desktop shortcuts are generated. This shortcuts start the teleop software and run the arm(s) and the hand(s).
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply README.md]: Fixed in the last commit

File: ansible/inventory/teleop/group_vars/control_machine.yml [hunk 4]
```diff
@@ -10,9 +10,26 @@ ethercat_left_arm: "enx2" # Ethercat interface of the left arm
...
 allow_auto_reboot: true # Allow Ansible to reboot the control_machine automatically during software installation
 use_aws: false # Boolean to indicate whether ROS logs will be uploaded to AWS S3
-glove: "haptx" # Determines which glove is installed (cyberglove or haptx)
+glove: "polhemus" # Determines which glove is installed (cyberglove, haptx or polhemus)
 cyberglove: "" # Specify the branch of sr_cyberglove_config for cyberglove configuration
 setup_directory: "/usr/local/bin" # Where setup scripts should be installed that are launched when Docker starts
 terminator: false # Launch terminator when the Docker starts
 nvidia_docker: false # Define if nvidia docker 2 is used (defaul is false)
 use_steamvr: false # Determines if SteamVR will launch inside the container created on this machine
+hand_side: "right" # Specify if the hand is right or left (skip for bimanual)
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply ansible/inventory/teleop/group_vars/control_machine.yml]: Fixed in the last commit. Yes, maybe we can do that in the future.

File: ansible/roles/products/teleop/server/desktop-icons/tasks/demohands-unimanual.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    hand_serial_value: "1370"
+    hand_side_value: "right"
+
+- name: Install desktop icon for hand D
+  import_tasks: nuc-control-loop-icon.yml
+  vars:
+    desktop_icon_png: "demohand_D_icon.png"
+    launch_script: "teleop_exec_D.sh"
+    config_branch_value: "demohand_D"
+    desktop_icon_name: "3 - Demohand D Launch NUC Right Side Teleop Hardware Control Loop"
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply ansible/roles/products/teleop/server/desktop-icons/tasks/demohands-unimanual.yml]: Fixed in the last commit

File: ansible/roles/products/teleop/server/desktop-icons/tasks/roslaunch-icon.yml [hunk 6]
```diff
@@ -14,21 +14,21 @@
...
     folder: "{{ shadow_hand_launcher_folder }}"
     shell_script_file_name: "{{ launch_script }}"
     icon_file_name: "{{ desktop_icon_png }}"
-    start_terminal: "true"
+    start_terminal: "false"
 
-- name: Make desktop icon for GUI icon trusted
-  shell: gio set "{{ user_folder }}/Desktop/{{ desktop_icon_name }}.desktop" "metadata::trusted" yes
+- name: Make desktop icon trusted
+  shell: gio set "{{ user_folder }}/Desktop/{{ desktop_icon_path }}.desktop" "metadata::trusted" yes
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply ansible/roles/products/teleop/server/desktop-icons/tasks/roslaunch-icon.yml]: There is no easy way to refresh. I looked into it, trying xdotool and wmctrl and restarting nautilus-desktop. I couldn't find a solution quickly. Maybe we can do it in the future (have a small separate user story).

File: ansible/roles/products/teleop/server/desktop-icons/templates/scripts/start-docker-container-nuc-interactive-terminal.j2 [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#jinja2: trim_blocks:False
+#!/bin/bash
+NUC_NAME={{ nuc_username }}
+NUC_ADDRESS={{ nuc_address }}
+CONTAINER={{ container_name }}
+ssh -Xt $NUC_NAME@$NUC_ADDRESS "docker start ${CONTAINER};docker exec -i --user user $CONTAINER bash -c 'cat > ~/.Xauthority' < ~/.Xauthority;docker exec -it --user user $CONTAINER /bin/bash -c $'source /home/user/projects/shadow_robot/base_deps/devel/setup.bash;source /home/user/projects/shadow_robot/base/devel/setup.bash;xauth list | tail -1 | awk \'{print \$1}\' | cut -d ':' -f2 > ~/.display_number;export DISPLAY=localhost:\$(<~/.display_number).0;bash'"
```
[the remark this answers is not in the document]
Comment (ToivoS) [reply ansible/roles/products/teleop/server/desktop-icons/templates/scripts/start-docker-container-nuc-interactive-terminal.j2]: We need this for the interactive terminal (https://stackoverflow.com/questions/42505339/why-use-t-with-ssh). This was the only way I could make it work.

File: bin/run-ansible.sh (hunk 1 above)
Comment (ToivoS) [reply bin/run-ansible.sh]: change the NUC password to the same password as delivery NUCs (I will tell you in chat)

**Pull request merged**